### ISSUE TITLE
[TS] error typing a simple closure returning method

### ISSUE THREAD

**Issue by javierfernandes** [uqbar-project/wollok#1167]
This came up by accident because I didn't remember wollok syntax for closures :S which I'm convinced is not intuitive at all coming from other languages like jscript/java/scala, and it is something really ambiguous like this case

This types fine
```wollok
object a {
	method m() { return true }
}
```
And it means **a method that returns a boolean** (could it be actually typed to `m() => TrueBooolean` ?.. anyway.. crazy idea)

This is also fine

```wollok
object a {
	method m() = true
}
```
And with the same semantic.

Now THIS raises an EValidator error (that one hiding an underlying issue on the TS)

```wollok
object a {
	method m() = { return true }
}
```
It just breaks it all and resolves to Any.
Of course (?) the semantic here is different, It is a method returning a Closure with no args that returns a Boolean.
Should be
```m() : Void => Boolean```

As a side note, this is a case where it could be difficult for a student to understand that a single '=' changed completely the semantic of what he had on the right side of it !!

**Comment by javierfernandes** [comment]
Disabling the TS avoids the issue.

**Comment by matifreyre** [comment]
> As a side note, this is a case where it could be difficult for a student
to understand that a single '=' changed completely the semantic of what he
had on the right side of it !!ç

Well... it's the usual thing, isn't it? Changing a single character CAN
change the whole meaning, and I would say that happens very frequently.

Which reminds me of a really short story (I know it in Spanish) that
highlights the importance of writing accurately and shows how a couple of
symbols can change something completely:

"Juan dijo: Pepe es un inútil."
vs.
"Juan, dijo Pepe, es un inútil."




On Fri, Jul 28, 2017 at 10:47 AM, Javier Fernandes <notifications@github.com

**Comment by javierfernandes** [comment]
Yeap, still the thing is that the grammar is not context-free on this one.

I ask you what is this ?
`{ return true }`

And you cannot know !

Depends on whether you put write that !

```
method a() { return true }     ==> it is a block of sentences
= { return true }                     ==> it is a closure
```

Of course here `const a = { return true }` could be easier to understand that it is a closure (still not so easy for me).  Here is even more confusing `method m() = { return true }` seems it ressembles a lot to `method m() { return true }`
It is not a matter of a change on a single character but the fact that they are ambiguous.

**Comment by PalumboN** [comment]
Fixed in v1.6.4
